File: data-agent/frontend/src/api/datasource.ts
import { api } from './auth'

export interface ColumnInfo {
  column_name: string
  column_type: string
  description: string
}

export interface DataSourceOut {
  id: number
  table_name: string
  original_filename: string
  description: string
  row_count: number
  columns: ColumnInfo[]
  created_at: string
}

export interface DataSourceListOut {
  sources: DataSourceOut[]
}

export async function getDataSources(): Promise<DataSourceOut[]> {
  const { data } = await api.get<DataSourceListOut>('/datasource/')
  return data.sources
}

export async function uploadDataSource(file: File): Promise<DataSourceOut> {
  const formData = new FormData()
  formData.append('file', file)
  const { data } = await api.post<DataSourceOut>('/datasource/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
  return data
}

export async function updateDataSource(
  sourceId: number,
  description: string,
  columnDescriptions: { column_name: string; description: string }[],
): Promise<DataSourceOut> {
  const { data } = await api.put<DataSourceOut>(`/datasource/${sourceId}`, {
    description,
    column_descriptions: columnDescriptions,
  })
  return data
}

export async function deleteDataSource(sourceId: number): Promise<void> {
  await api.delete(`/datasource/${sourceId}`)
}

File: data-agent/backend/app/schemas/datasource.py
"""数据源相关 Schema。"""
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel


class ColumnInfo(BaseModel):
    column_name: str
    column_type: str
    description: str = ""


class DataSourceOut(BaseModel):
    id: int
    table_name: str
    original_filename: str
    description: str
    row_count: int
    columns: List[ColumnInfo]
    created_at: datetime

    class Config:
        from_attributes = True


class DataSourceUpdate(BaseModel):
    description: Optional[str] = None
    column_descriptions: Optional[List[dict]] = None  # [{"column_name": "xxx", "description": "xxx"}]


class DataSourceListOut(BaseModel):
    sources: List[DataSourceOut]
